experimental/dds/tree2/docs/schema2.md

```diff
@@ -0,0 +1,129 @@
+5. It requires use of symbols to avoid colliding with local field names in many places in the API, which has been confusing users (many TypeScript developers don't even know what Symbols are).
+6. Supporting all the things in all the nodes makes APIs complex. This is extra annoying since this complicates the schema-aware API.
+7. Extra fields require proxies to implement the desired TypeScript APIs for nodes.
+8. Having both global and local fields adds complexity, particularly to schema and path storage where symbols can't be used.
+9. Supporting extraLocalFields and extraGlobalFields on nodes while providing a JavaScript object like API requires a Proxy, which adds overhead and complicates code implementation and maintenance.
+
+## Proposal
+
+Instead of one kind of node in view schema, have 4, each with a subset of our current functionality:
+
+1. Terminal Node: holds a value, but no children.
+   Don't support `undefined` as a value: an empty Struct can be used for that case if needed.
+2. Struct Node: finite list of fields (key+field type).
+3. Map Node: node with a single single field type which must permit empty.
+   Allows any field of that type under any string field key.
+   Provides the functionality currently done by making a node schema with extra local fields.
+4. Field Node: Has a single unnamed filed (using the empty field key).
+   When reading in the editable tree API implicitly, unwraps to the field.
+   This provides the functionality currently done by making a node schema with a primary field (empty field key), which used for "array nodes".
+
+Each of these will get a separate API to declare (similar to SchemaBuilder.object and SchemaBuilder.primitive, except we will have 4 options).
+Each will also get its own API for accessing it in editable tree, though all will extend a common generic tree traversal API as well.
+
+This will mirror how field kinds work, where we have a small collection of supported kinds in schema, and each gets its own editable tree API.
+
+Struct Nodes will also get an additional feature: custom field names.
+Custom field names allow the view schema to declare the string to be used for the name of the field in schema-aware APIs instead of just using the field key.
+This can default to the field key, but can optionally be a distinct string.
+
+Customizable field names are great for app maintainability: a field in coded can be renamed as used in code without breaking existing data.
```

Comment:
There's something wonky with the "a field in coded can be renamed as used in code" part